src/CPUSolver.cpp

```diff
@@ -652,16 +649,9 @@ FP_PRECISION CPUSolver::computeFSRSources() {
     else
       fission_source = 0.0;
 
-    /* Compute total scattering source for group G */
     for (int G=0; G < _num_groups; G++) {
-      scatter_source = 0;
-
-      for (int g=0; g < _num_groups; g++)
-        _scatter_sources(tid,g) = material->getSigmaSByGroupInline(g,G)
-                      * _scalar_flux(r,g);
-
-      scatter_source=pairwise_sum<FP_PRECISION>(&_scatter_sources(tid,0),
-                                                _num_groups);
+      /* Compute total scattering source for group G */
+      scatter_source = material->getScatterSource(G,&_scalar_flux(r,0));
```

Comment:
For uniformity, might it be a good idea to do the same thing for the fission source - add a ``Material::getFissionSource(group, flux)`` routine? Even if this isn't something you foresee using in the future for your self-shielding methods, it might make the code easier to follow.